lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -99,6 +99,113 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                       binder.op, resultType, lhs, rhs);
                   return success();
                 });
+  patterns.onOp(
+      "MaxPool", 12, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        std::string autoPad;
+        SmallVector<int64_t> dilations;
+        if (binder.customOpNameStringAttr(autoPad, "auto_pad", "NOTSET"))
+          return failure();
+        if (autoPad != "NOTSET") {
+          // TODO: Add support for `auto_pad` != "NOTSET"
+          return rewriter.notifyMatchFailure(
+              binder.op, "unsupported conversion: auto_pad != NOTSET");
+        }
+        if (binder.s64IntegerArrayAttr(dilations, "dilations", {})) {
+          return failure();
+        }
+
+        Torch::ValueTensorType resultType;
+        Value operand;
+        bool ceilMode;
+        int64_t storageOrder;
+        // TODO: Add support for indices output and storage_order
```

Comment:
I would have expected that we would have failed the match on an unsupported storage order and/or indices output as the op is most likely invalid. Is this not so?

Reply:
I didn't add the support for this too because from the test files generated, there is no test for this 2. And for resnet support for now, it probably not necessary. And I haven't figured out a way to setup a storageorder in torch_mlir tensor.

Reply:
I think you need to still need to report a match failure if the value for `storage_order` is not the expected value. I think the check below validates that we get the value but does not alter behaviour depending on its value.